storage/series/series.go

```diff
@@ -108,27 +117,35 @@ func (s *dbSeries) Tick() (TickResult, error) {
 	// In best case when explicitly asked to drain may have no
 	// stale buckets, cheaply check this case first with a Rlock
 	s.RLock()
+
 	needsDrain := s.buffer.NeedsDrain()
 	needsBlockUpdate := s.needsBlockUpdateWithLock()
 	r.ActiveBlocks = s.blocks.Len()
+
 	s.RUnlock()
 
 	if !needsDrain && !needsBlockUpdate {
 		return r, nil
 	}
 
 	s.Lock()
-	// Recalculate in case changed during lock promotion
-	needsDrain = s.buffer.NeedsDrain()
-	needsBlockUpdate = s.needsBlockUpdateWithLock()
+
+	// NB(r): don't bother to check if needs drain or needs
+	// block update still holds as running both these checks
+	// are relatively expensive and running these methods
+	// will be a no-op in case conditions no longer hold.
 	if needsDrain {
```

Comment:
That means if `needsDrain` changes from false to true in between locking, we won't drain and reset it, is that ok?